fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -471,9 +471,18 @@ public PlanFragment visitPhysicalHashJoin(
                 rightChildOutputTupleDesc != null ? rightChildOutputTupleDesc : context.getTupleDesc(rightPlanRoot);
         TupleDescriptor outputDescriptor = context.generateTupleDesc();
         Map<ExprId, SlotReference> slotReferenceMap = Maps.newHashMap();
-        hashJoin.getOutput().stream()
-                .map(SlotReference.class::cast)
-                .forEach(s -> slotReferenceMap.put(s.getExprId(), s));
+
+        if (hashJoin.getOtherJoinCondition().isPresent()) {
```

Comment:
we need to distinguish between intermediate tuple and output tuple. just intermediate tuple need expand slot to both side of child. output tuple should follow the original logic.